src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -207,9 +210,12 @@ class MarathonSchedulerActor private (
     case msg => logger.warn(s"Received unexpected message from ${sender()}: $msg")
   }
 
-  def scaleRunSpecs(): Unit = {
-    groupRepository.root().foreach { root =>
-      root.transitiveRunSpecs.foreach(spec => self ! ScaleRunSpec(spec.id))
+  def scaleRunSpecs(): Future[Done] = {
```

Comment:
I'm not sure this should be a future in this diff. We can follow up but the pull request should be about changing `Future.sequence` to a stream.

Also, why was this changed from a tell to an ask?

Reply:
`tell` = no backpressure, `ask` will provide some kind of feedback